### PR TITLE
Add API stub for real time recency

### PR DESCRIPTION
**Pull request by jkosh44**
### Motivation
This PR adds a known-desirable feature.

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  -

### CODE REVIEW

File: src/controller/src/lib.rs [hunk 1]
```diff
@@ -213,6 +213,14 @@ where
             }
         }
     }
+
+    /// Produces a timestamp that reflects all data available in
+    /// `source_ids` at the time of the function call.
+    #[allow(unused)]
+    #[allow(clippy::unused_async)]
+    pub async fn recent_timestamp(&self, source_ids: Vec<GlobalId>) -> T {
```
Comment (benesch) [inline src/controller/src/lib.rs:221]: Hard to say at the moment, but this probably can take a slice instead of a vec?

**Review by benesch — COMMENTED** [review]
I love this PR.

File: src/controller/src/lib.rs (hunk 1 above)
Comment (jkosh44) [reply src/controller/src/lib.rs]: Just switched to a slice.

**Review by petrosagg — APPROVED** [review]
Nice

**Pull request merged**